src/components/FileHistoryElement.tsx

```diff
@@ -14,85 +18,125 @@ export function FileHistoryElement(props: props) {
 
   let fileCommits: GitLogEntry[] = []
   if (props.clickedObject.type === "blob") {
-    fileCommits = props.clickedObject.commits.map((c) => analyzerData.commits[c])
+      fileCommits = props.clickedObject.commits.map((c) => analyzerData.commits[ c ])
   } else {
-    try {
       fileCommits = Array.from(calculateCommitsForSubTree(props.clickedObject))
-        .map((c) => analyzerData.commits[c])
+        .map((c) => analyzerData.commits[ c ])
         .sort((a, b) => b.time - a.time)
-    } catch (e) {
-      console.log(e)
-    }
   }
 
-  return <CommitHistory commits={fileCommits} />
+  return <CommitHistory commits={ fileCommits } />
 }
 
 interface CommitDistFragProps {
   items: GitLogEntry[]
   show: boolean
+  commitCutoff: number
+  sortBy?: SortCommitsMethods
+  handleOnClick?: (commit: GitLogEntry) => void
 }
 
 export function CommitDistFragment(props: CommitDistFragProps) {
-  if (!props.show) return null
+  if (!props.show || !props.items) return null
+  const sortMethod: SortCommitsMethods = props.sortBy !== undefined ? props.sortBy : "date"
+
+  const cleanGroupItems: { [ key: string ]: GitLogEntry[] } = sortCommits(props.items, sortMethod)
+
+  const items: Array<AccordionData> = new Array<AccordionData>()
+  for (const [ key, values ] of Object.entries(cleanGroupItems)) {
+    items.push({
+      title: key,
+      content: (
+        <>
+          { values.map((value: GitLogEntry) => {
+            return (
+              <>
+                <li
+                  className="cursor-auto"
+                  style={{listStyleImage: `url(${commitIcon})`}}
+                  onClick={ () => (props.handleOnClick ? props.handleOnClick(value) : null) }
+                  key={ value.time + value.message + "--itemContentAccordion" }
+                >
+                  { value.message }
+                </li>
+              </>
+            )
+          }) }
+        </>
+      ),
+    })
+  }
 
   return (
     <>
```

Comment:
This outer fragment is redundant since it only contains a single child.

Reply:
Done 🙂